app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
3. That means the user can't use actual output in the planning phase. It means dynflow has the liberty to forbid actual calls to other tasks. All it has to supply is an interface to say that current task depends on other one. Example for such approach:

``` Ruby
#current state
def plan()
  x = other_task.output[:foo][:get_number] #x will be set to an OutputReference.
  # the fact that this task has a dependency on other_task has been recorded.
  y = x + 1 #will fail, since x is not a number at this point.
end

# suggested state
def plan()
  other_task_state = other_task.result #other_task_result will be set to an OutputReference
  #or maybe even
  other_task_state = add_dependency(:other_task)

  # the fact that this task has a dependency is recorded
  other_task_state.foo # is forbidden - it's a useless call here, since the result will be available only in run phase
end
```

This way we eliminate the need to record call chains for future tasks.